azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/WebAppBase.java

```diff
@@ -216,6 +216,12 @@
      */
     String autoSwapSlotName();
 
+    /**
+     * @return the managed service identity in Azure Active Directory for this web app
+     */
+    @Beta(SinceVersion.V1_6_0)
+    ManagedServiceIdentity managedServiceIdentity();
```

Comment:
One suggestion here is: It would be nice to be more explicit here. Instead of returning `ManagedServiceIdentity` we could have two separate properties 

`string systemAssignedManagedServiceIdentityTenantId()`, 
`string systemAssignedManagedServiceIdentityPrincipalId()`. 

This will guard us from making breaking change when web app eventually supports `user assigned` managed service identity.  Also ensures consistency in MSI interface across fluent resources (VM, VMSS, Storage).

Btw just FYI we also enabled `isManagedServiceIdentityEnabled()` getter in other resources (VM, VMSS, Storage) that supports MSI

Reply:
Also in the gen-code there is this property `object ManagedServiceIdentity::type` of type object. I think we should ask web-app team to be specific here in swagger, today the supported value for type in case of other RPs are `SystemAssigned`, `UserAssigned`, `SystemAssigned,UserAssigned` and `None`.

Reply:
Yes the Object is actually just a string. I'll look at those resources and try to make it consistent.